lib/usecases/login_use_case.dart

```diff
@@ -0,0 +1,34 @@
+import 'dart:async';
+
+import 'package:flutter/cupertino.dart';
+import 'package:survey/exception/network_exceptions.dart';
+import 'package:survey/models/auth_token.dart';
+import 'package:survey/repositories/oauth_repository.dart';
+import 'package:survey/usecases/base_use_case.dart';
+
+class LoginCredential {
+  @required
+  final String email;
+  @required
+  final String password;
+
+  LoginCredential(this.email, this.password);
+}
+
+class LoginUseCase extends UseCase<AuthToken, LoginCredential> {
+  final OAuthRepository _repository;
+
+  const LoginUseCase(this._repository);
+
+  @override
+  Future<Result<AuthToken>> call(LoginCredential credential) async {
+    try {
+      final value =
+          await _repository.login(credential.email, credential.password);
+      return Success(value);
+    } catch (e) {
+      final apiError = ApiError(NetworkExceptions.getDioException(e), e);
+      return Future.value(Failed(apiError));
```

Comment:
Hmm I think we only need `return Failed(apiError)`, right? 🤔

Reply:
it receives Future<> as the return type 🥴

Reply:
I did create a test to make sure, so they both give the same result, but I prefer to just `return Failed(apiError)` cause we mark it as `async`. What do you think?
```dart
Future<int> test() async {
  return 2;
}
Future<int> test2() async {
  return Future.value(2);
}

void main() async {
  print(await(test())); // 2
  print(await(test2())); // 2
}
```

Reply:
ohhh...I think I added async lately and changed the positive case but forgetting to convert the catch case 😳  thanks for noticing, let me retry and will update in the next PR

Reply:
yep it works, will update it 🙇  at first I tried to reduce the try/catch with chained functions if possible...but the function chains here is kinda weird I think I will spend more time on it later 🙇 
![image](https://user-images.githubusercontent.com/13435717/112785246-c37d1b80-907d-11eb-95e5-82c994e93bc2.png)